app/main/lib/text_similarity.py

```diff
@@ -202,3 +210,8 @@ def search_text_by_model(search_params):
     return {
         'result': response
     }
+def retrieve_openai_embeddings(text, model_key):
+    openai.api_key = app.config['OPENAI_API_KEY']
+    app.logger.info(f"Calling OpenAI API")
+    model_key_without_openai_prefix = model_key[len(PREFIX_OPENAI):]
+    return openai.embeddings_utils.get_embedding(text, engine=model_key_without_openai_prefix)
```

Comment:
@ahmednasserswe We should handle the case where this line fails. E.g., if the API is down, API key is invalid, etc. 
It should fail gracefully, log the error, and other models should continue.